auxx/src/Command/Update.hs

```diff
@@ -113,24 +113,31 @@ propose sendActions ProposeUpdateParams{..} = do
     updateData <- mapM updateDataElement puUpdates
     let udata = HM.fromList updateData
     let whenCantCreate = error . mappend "Failed to create update proposal: "
-    withSafeSigner skey (pure emptyPassphrase) $ \case
-        Nothing -> putText "Invalid passphrase"
-        Just ss -> do
-            let updateProposal = either whenCantCreate identity $
-                    mkUpdateProposalWSign
-                        puBlockVersion
-                        bvm
-                        puSoftwareVersion
-                        udata
-                        (mkAttributes ())
-                        ss
-            if null ccPeers
-                then putText "Error: no addresses specified"
-                else do
-                    submitUpdateProposal (immediateConcurrentConversations sendActions ccPeers) ss updateProposal
-                    let id = hash updateProposal
-                    putText $
-                      sformat ("Update proposal submitted, upId: "%hashHexF) id
+
+    skeys <- if not puVoteAll then pure [skey]
+             else getSecretKeysPlain
+    withSafeSigners skeys (pure emptyPassphrase) $ \ss -> do
+        unless (length skeys == length ss) $
+            error $ "Number of safe signers: " <> show (length ss) <> ", expected " <> show (length skeys)
+        let publisherSS = ss !! if not puVoteAll then 0 else puIdx
```

Comment:
AFAIU, `publisherSS` is the same as `skey`. Do we need two different variables?

Reply:
lol `skey` is secret key, publisherSS is safe signer